runtime/compiler/arm/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -479,27 +479,22 @@ uint8_t *J9::ARM::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterated
 
          cursor = self()->emitClassChainOffset(cursor, inlinedCodeClass);
 
-         uintptrj_t offset;
-         TR::MethodSymbol *calleeSymbol = callSymRef->getSymbol()->castToMethodSymbol();
-         if (!TR::Compiler->cls.isInterfaceClass(comp, inlinedCodeClass) && calleeSymbol->isInterface())
+         J9Method *inlinedJ9Method = reinterpret_cast<J9Method *>(inlinedMethod->getNonPersistentIdentifier());
+         J9Method *resolvedMethods = static_cast<J9Method *>(fej9->getMethods(inlinedCodeClass));
+
+         uint32_t numMethods = fej9->getNumMethods(inlinedCodeClass);
+         uint32_t methodIndex = 0;
+         for (; methodIndex < numMethods ; methodIndex++)
```

Comment:
It bothers me a little bit that this is looping when a pointer subtraction would do. Unfortunately, pointer subtraction is UB in case some bug causes `inlinedJ9Method` to be missing from `resolvedMethods`. I think it's possible to do the subtraction using integers and check the range though, e.g.
```
uintptr_t methodOffset = (uintptr_t)inlinedJ9Method - (uintptr_t)resolvedMethods;
TR_ASSERT_FATAL(methodOffset % sizeof (J9Method) == 0, "...");
uintptr_t methodIndex = methodOffset / sizeof (J9Method);
TR_ASSERT_FATAL(methodIndex < numMethods, "...");
// Now it's safe to truncate methodIndex to 32-bit
```
But I understand if you're more comfortable with the loop :man_shrugging:

Reply:
I don't have a concern with not using a loop as long as there's i add a comment describing what the code's doing.